app/router.js

```diff
@@ -6,7 +6,10 @@ var Router = Ember.Router.extend({
   location: config.locationType,
   onTransition: function() {
     this.get('analytics').page();
-    this.get('flashMessages').clearMessages();
+
+    if(config.flashMessageDefaults.sticky) {
```

Comment:
This code is correct.

The default behavior, `sticky: false`, has the flashMessages disappear after a timeout. This is useful because the messages are visible after a transition, so it's very clear when a new item is created or updated.

However, this timeout breaks the `andThen` behavior in tests, so `sticky` is set to true during testing.